include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -345,29 +345,57 @@ typedef enum OrtCudnnConvAlgoSearch {
 */
 typedef struct OrtCUDAProviderOptions {
 #ifdef __cplusplus
-  OrtCUDAProviderOptions() : device_id{}, cudnn_conv_algo_search{OrtCudnnConvAlgoSearchExhaustive}, gpu_mem_limit{SIZE_MAX}, arena_extend_strategy{}, do_copy_in_default_stream{}, has_user_compute_stream{}, user_compute_stream{}, default_memory_arena_cfg{} {}
+  OrtCUDAProviderOptions() : device_id{}, cudnn_conv_algo_search{OrtCudnnConvAlgoSearchExhaustive}, gpu_mem_limit{SIZE_MAX}, arena_extend_strategy{}, do_copy_in_default_stream{1}, has_user_compute_stream{}, user_compute_stream{}, default_memory_arena_cfg{} {}
 #endif
 
-  int device_id;  ///< CUDA device id (0 = default device)
+  /** \brief CUDA device Id
+  *   Defaults to 0.
+  */
+  int device_id;
+
+  /** \brief CUDA Convolution algorithm search configuration.
+  *   See enum OrtCudnnConvAlgoSearch for more details.
+  *   Defaults to OrtCudnnConvAlgoSearchExhaustive.
+  */
   OrtCudnnConvAlgoSearch cudnn_conv_algo_search;
 
   /** \brief CUDA memory limit (To use all possible memory pass in maximum size_t)
-  *
-  * \note If a ::OrtArenaCfg has been applied, it will override this field
+  *   Defaults to SIZE_MAX.
+  *   \note If a ::OrtArenaCfg has been applied, it will override this field
   */
   size_t gpu_mem_limit;
 
   /** \brief Strategy used to grow the memory arena
-  *
-  * 0 = kNextPowerOfTwo<br>
-  * 1 = kSameAsRequested<br>
-  * \note If a ::OrtArenaCfg has been applied, it will override this field
+  *   0 = kNextPowerOfTwo<br>
+  *   1 = kSameAsRequested<br>
+  *   Defaults to 0.
+  *   \note If a ::OrtArenaCfg has been applied, it will override this field
   */
   int arena_extend_strategy;
+
+  /** \brief Flag indicating if copying needs to take place on the same stream as the compute stream in the CUDA EP   
+  *   0 = Use separate streams for copying and compute.
+  *   1 = Use the same stream for copying and compute.
+  *   Defaults to 1.
+  *   WARNING: Setting this to 0 may result in data races for some models.
```

Comment:
If we want this to be 1, ideally, we shouldn't even give the user an option to configure this. But we can't change the struct now. So, we should at least enforce this or else it'll lead to support issues. Can be done in a separate PR.

Reply:
@KeDengMS added this field with the intention of giving the user an option.